src/main/java/com/salesforce/mirus/MirusSourceTask.java

```diff
@@ -148,8 +148,11 @@ private void seekToOffsets(List<TopicPartition> partitionIds) {
             }
             return;
           }
-          long lastRecordedOffset = (long) offsetMap.get(KEY_OFFSET);
-          consumer.seek(tp, lastRecordedOffset);
+          Long lastRecordedOffset = (Long) offsetMap.get(KEY_OFFSET);
+          // check if offset has been set to null, i.e. tombstone record
+          if (lastRecordedOffset != null) {
```

Comment:
This case should be handled exactly the same way as seeing no offsets as all.  I think you just need to add a new clause triggering the `if (offsetMap ==null)` logic above - so probably `if (offsetMap ==null || offsetMap.get(KEY_OFFSET) == null)`.  You will need to rearrange slightly.  When complete we should seek to latest if "auto.offset.reset" has been set to latest (which it is not by default).

Reply:
Good point. I was thinking only about discarding/ignoring a topic/partition. But there are cases such as considering a topic/partition after being discarded. So yes, we should be treating non-existent and existent while equal to null the same way.